### ISSUE TITLE
Initial browser loading takes too long

### ISSUE THREAD

**Issue by MageSlayer** [piroor/treestyletab#2542]
## Short description
Firefox loads for several minutes (!) when several hundred tabs exist.

## Steps to reproduce
Open or better generate several hundred tabs and restart browser.

## Expected result
Only visible (not collapsed) tabs are taken into account. That is lazy display expected.

## Actual result
I reckon some non-linear algorithms are used, so it takes so unproportionally long.

## Environment

 * Platform (OS): Linux x64
 * Version of Firefox: 68.6.0
 * Version (or revision) of Tree Style Tab: 2.8.7

**Comment by piroor** [comment]
I also think your TST version is too old. The latest TST still supports ESR68. If this problem depends on exact the old version and you cannot update to a newer one, I think there is nothing I can do...

**Comment by MageSlayer** [comment]
Hm.
Looks like updating was disabled for some reason.
I'll retest using newest version.
Thanks for hints.

**Comment by MageSlayer** [comment]
Can't reproduce with latest version.
I guess it might be closed.

**Comment by p0358** [comment]
Reopen please. I am on 3.5.1 version from 3 days ago and this happens to me. 888 tabs (for example) and it starts for minutes. Also it is somehow breaking the browser, other extenions are broken until I restart them, but until Tree Style Tab is completely loaded, I see that I cannot switch tabs with the top bar at all, or I switch them, but the page doesn't load, and for extensions, toggling their state on about:addons has no effect, RAM usage can jump up to 20 GB (then falls down to less than 2 once loading is done).
Disabling TST makes these issues go extinct, and restoring browser with all these tabs takes up to a few seconds.
Also I feel like enabling "Tab flip for Tree Style Tab" addon makes this even worse, but the issue exists and is very noticeable even with just TST alone.
My platform: Windows 10 1909
My Firefox version: 76.0b8

**Comment by piroor** [comment]
@p0358 Reopening a closed issue will entangle the initial reporter (and other commenters) of the issue even if the problem doesn't happen on their environment anymore. Could you post a new issue? And, hopefully could you investigate more for debugging? Current report has too less information to make the bottleneck clear by me.

**Comment by MageSlayer** [comment]
Hi again.

After upgrading to recent version, the issue seems to be gone.
However, I suspect that it might be related to my sudden "flattening" of all tabs. I mean before some crash, I had deep tree of tabs. But after restart all tabs became a linear list. 

So my point is the issue might be related to deep trees.
